scipy/linalg/tests/test_matfuncs.py

```diff
@@ -728,6 +729,24 @@ def test_readonly(self):
         a.flags.writeable = False
         expm(a)
 
+    def test_gh18086(self):
+        A = np.zeros((400, 400), dtype=float)
+        i = np.random.default_rng(0).integers(0, 399, 500)
+        j = np.random.default_rng(100).integers(0, 399, 500)
+        A[i, j] = np.random.default_rng(1000).random(500)
```

Comment:
This pattern of multiple default_rngs with varying seeds is a bit unusual. Is there a reason to not use a single `rng=np.random.default_rng(some seed) ` and drawing from it throughout the test?

Reply:
Good point! I consolidated the RNGs and picked a seed that gave me `m=9` locally. Hopefully that isn't platform dependent.